src/stirling/source_connectors/socket_tracer/conn_trackers_manager.cc

```diff
@@ -109,7 +110,18 @@ uint64_t GetConnMapKey(uint32_t pid, int32_t fd) { return (static_cast<uint64_t>
 
 }  // namespace
 
-ConnTrackersManager::ConnTrackersManager() : trackers_pool_(kMaxConnTrackerPoolSize) {}
+ConnTrackersManager::ConnTrackersManager()
+    : trackers_pool_(kMaxConnTrackerPoolSize),
+      total_conn_trackers_(
+          BuildGauge("total_conn_trackers", "Total number of conn trackers in use")),
+      ready_for_destruction_(BuildGauge("ready_for_destruction",
```

Comment:
I suspect this gauge might be a bit noisy. My understanding is that ready-for-destruction conn trackers are cleaned up relatively quickly (at least on a time scale of seconds rather than minutes). At the moment, prometheus metrics are only sampled once a minute, so given this gauge is measuring something that's timescale is much smaller than that, it might be pretty noisy. Depending on what you want to use the number for, it might be more useful to keep a counter of the high water mark of conn trackers that are ready for destruction. Perhaps, you might also want a high water mark for total conn trackers, although I think sampling total conn trackers at once a minute is much more reasonable.

Reply:
I agree with your concern and I didn't have immediate plans for the two gauge metrics added in this PR. The counter metrics are the ones I plan to use to validate the tls changes I'm working on. I decided to instrument the gauge metrics since it filled out what we track with our internal stats interface. However, maybe we should just omit these for now since it may be noisy.

Is there any way to filter out metrics from the prometheus scraping side? Wondering if it's possible to have the instrumentation there but enable it later if we decided it was valuable.

Reply:
We don't currently have a good way to filter them out on the scraping side. If you think you might use them, I think its fine to leave them in, otherwise I would just remove the instrumentation for now.

Reply:
I went ahead and removed them since I don't see any concrete need for now.